pkg/epp/backend/metrics/metrics.go

```diff
@@ -37,15 +37,13 @@ const (
 )
 
 type PodMetricsClientImpl struct {
-	MetricMapping *MetricMapping
+	MetricMapping          *MetricMapping
+	ModelServerMetricsPort int32
+	ModelServerMetricsPath string
 }
 
 // FetchMetrics fetches metrics from a given pod, clones the existing metrics object and returns an updated one.
-func (p *PodMetricsClientImpl) FetchMetrics(ctx context.Context, pod *backend.Pod, existing *MetricsState, port int32) (*MetricsState, error) {
-	// Currently the metrics endpoint is hard-coded, which works with vLLM.
-	// TODO(https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/16): Consume this from InferencePool config.
-	url := "http://" + pod.Address + ":" + strconv.Itoa(int(port)) + "/metrics"
-
+func (p *PodMetricsClientImpl) FetchMetrics(ctx context.Context, pod *backend.Pod, existing *MetricsState, url string) (*MetricsState, error) {
```

Comment:
seems to me that `GetMetricEndpoint` and `FetchMetrics` calls always come together (one must calculate the url before calling `FetchMetrics`).
does it make sense to make `GetMetricEndpoint` function private, remove the url arg from the `FetchMetrics` function, and get the url inside FetchMetrics (for example in the first line of the function)?
(and then GetMetricEndpoint function can also be removed from the interface)